gemrb/core/Scriptable/Scriptable.cpp

```diff
@@ -2243,14 +2243,16 @@ bool Movable::DoStep(unsigned int walkScale, ieDword time) {
 		dy = std::min(dy * stepTime / walkScale, dyOrig);
 		dx = std::ceil(dx) * xSign;
 		dy = std::ceil(dy) * ySign;
-		Actor* actorInTheWay = area->GetActor(Point(Pos.x + 2 * dx, Pos.y + 2 * dy), GA_NO_DEAD|GA_NO_UNSCHEDULED|GA_NO_SELF, this);
-		if (actorInTheWay) {
+		const float COLLISION_LOOKAHEAD = 4;
+		Actor* actorInTheWay = area->GetActorInRadius(Point(Pos.x + COLLISION_LOOKAHEAD * dx, Pos.y + COLLISION_LOOKAHEAD * dy),
+				GA_NO_DEAD|GA_NO_UNSCHEDULED, COLLISION_LOOKAHEAD);
+		if (actorInTheWay && actorInTheWay != this) {
```

Comment:
You can add GA_NO_SELF above and then the sameness check will not be needed any more.

Reply:
or is it another microoptimisation?

Reply:
`GetActorInRadius` doesn't support `GA_NO_SELF` because it has no `checker` argument. I can just add it though and let it default to `NULL`

Reply:
Ah, never mind then.